front/src/applications/stdcm/hooks/useStaticPathfinding.ts

```diff
@@ -20,56 +16,45 @@ const useStaticPathfinding = (infra?: InfraWithState) => {
   const pathSteps = useSelector(getStdcmPathSteps);
   const { rollingStock } = useStoreDataForRollingStockSelector();
 
-  const [pathfinding, setPathfinding] = useState<PathfindingResult>();
-
-  const [postPathfindingBlocks] =
-    osrdEditoastApi.endpoints.postInfraByInfraIdPathfindingBlocks.useLazyQuery();
-
   const pathStepsLocations = useMemo(
     () => compact(pathSteps.map((step) => step.location)),
     [pathSteps]
   );
 
-  const pathProperties = usePathProperties(
-    infra?.id,
-    pathfinding?.status === 'success' ? pathfinding : undefined,
-    ['geometry']
-  );
-
-  useEffect(() => {
-    const launchPathfinding = async () => {
-      setPathfinding(undefined);
-      if (infra?.state !== 'CACHED' || !rollingStock || pathStepsLocations.length < 2) {
-        return;
-      }
-
-      const payload = getPathfindingQuery({
-        infraId: infra.id,
-        rollingStock,
-        pathSteps: pathStepsLocations,
-      });
-
-      if (payload === null) {
-        return;
-      }
-
-      const pathfindingResult = await postPathfindingBlocks(payload).unwrap();
-
-      setPathfinding(pathfindingResult);
-    };
-
-    launchPathfinding();
+  const pathfindingPayload = useMemo(() => {
+    if (infra?.state !== 'CACHED' || !rollingStock || pathStepsLocations.length < 2) {
+      return undefined;
+    }
```

Comment:
The other checks are done in `getPathfindingQuery`. And to match `getPathfindingQuery` return, maybe it' better to return also `null` if infra state is not cached.
```suggestion
    if (infra?.state !== 'CACHED') {
      return null;
    }
```